mailet/standard/src/main/java/org/apache/james/transport/mailets/LogMessage.java

```diff
@@ -52,18 +52,81 @@
      * or mark it as finished.
      */
     private final Logger logger;
+    private final LogLevel logLevel;
     private boolean passThrough = true;
-    private boolean headers = true;
-    private boolean body = true;
-    private int bodyMax = 0;
-    private String comment = null;
+    private boolean headers;
+    private boolean body;
+    private int bodyMax;
+    private String comment;
+    private List<String> specificHeaders;
+    private List<String> specificAttributes;
+
+    private StringBuffer logBuffer = new StringBuffer();
+
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private Logger logger;
+        private List<String> specificHeaders;
+        private List<String> specificAttributes;
+        private boolean headers;
+        private boolean body;
+        private LogLevel logLevel;
+
+        private Builder(){
+            logger = LOGGER;
+            specificHeaders = new LinkedList<>();
+            specificAttributes = new LinkedList<>();
+            body = false;
+            headers = false;
+        }
+
+        public Builder logger(final Logger logger) {
+            this.logger = logger;
+            return this;
+        }
+
+        public Builder logLevel(final LogLevel logLevel) {
+            this.logLevel = logLevel;
+            return this;
+        }
+
+        public Builder specificHeaders(List<String> headers) {
+            this.specificHeaders.addAll(headers);
+            return this;
+        }
+
+        public Builder specificAttributes(List<String> attributes) {
+            this.specificAttributes.addAll(attributes);
+            return this;
+        }
+
+        public Builder hasHeaders(boolean headers) {
+            this.headers = headers;
+            return this;
+        }
+
+        public Builder hasBody(boolean body) {
+            this.body = body;
+            return this;
+        }
+
+        public LogMessage build(){
+            return new LogMessage(logger,logLevel,specificHeaders,specificAttributes,headers,body);
+        }
 
-    public LogMessage(Logger logger) {
-        this.logger = logger;
     }
 
-    public LogMessage() {
-        this(LOGGER);
+    private LogMessage(Logger logger, LogLevel logLevel, List<String> specificHeaders, List<String> specificAttributes, boolean headers, boolean body){
```

Comment:
While having a builder is clearly valuable, James needs to be able to 'inject' fields in that mailet.

So, we need to read the LogLevel, specificHeaders, specificAttributes headers & body parameters from the configuration, in the `init` method.